### ISSUE TITLE
Test report ignore junit5 display name annotation for test class

### ISSUE THREAD

**Issue by rosolko** [gradle/gradle#4423]
### Expected Behavior

Show also value from `@DisplayName` annotation in tests result report.
From example it's - *Test class display name*

### Current Behavior

Report show me test class name.

<img width="725" alt="att_1" src="https://user-images.githubusercontent.com/11458319/36351112-9f832a04-14b5-11e8-83c6-8244adc244d2.PNG">
<img width="521" alt="att_2" src="https://user-images.githubusercontent.com/11458319/36351113-9fc411ae-14b5-11e8-876f-9bedcc95d3da.PNG">

### Steps to Reproduce (for bugs)

```
@DisplayName("Test class display name")
class SampleTests {

    @Test
    @DisplayName("Test display name")
    void SampleTest() {
        assertTrue(true);
    }
}
```

### Your Environment

```
------------------------------------------------------------
Gradle 4.6-rc-1
------------------------------------------------------------

Build time:   2018-02-16 16:28:07 UTC
Revision:     ac6267d33138856ffe71b159ab8154e71e9da72e

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          1.8.0_162 (Oracle Corporation 25.162-b12)
OS:           Windows 10 10.0 amd64
```

**Comment by blindpirate** [comment]
Thanks for reporting, this is a known issue. Because the test structure is based on test class name, currently we didn't make `@DisplayName` on class displayed. I'll try to make this into `4.6`.

**Comment by blindpirate** [comment]
Closed by #4425
